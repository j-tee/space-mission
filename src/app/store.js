/* eslint-disable */

import { configureStore } from '@reduxjs/toolkit';

export  const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});
