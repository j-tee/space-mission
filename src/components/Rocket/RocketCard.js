/* eslint-disable camelcase */
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { cancelRocket, reserveRocket } from '../../redux/rocket/rocketSlice';

const RocketCard = ({ rocket }) => {
  const dispatch = useDispatch();
  // Destructure rocket keys
  const {
    id, active, flickr_images, description, rocket_name,
  } = rocket;

  // Reserve rocket handler
  const handleReserve = (e) => {
    const data = Number(e.target.id);
    dispatch(reserveRocket(data));
  };

  // Cancel rocket reserve
  const handleCancelReserve = (e) => {
    const data = Number(e.target.id);
    dispatch(cancelRocket(data));
  };

  return (
    <div className="rocket-card">
      <img className="image" src={flickr_images} alt="rocket" />
      <div className="rocket-desc">
        <h1 className="title-desc">{rocket_name}</h1>
        <p className="info">
          {active === true && (<span className="rocket-reserve">Reserved</span>)}
          {description}
        </p>
        { active === true
          ? (
            <button className="cancel-btn btn" type="button" id={id} onClick={handleCancelReserve}>Cancel Reservation </button>
          )
          : (
            <button className="reserve-btn btn" type="button" id={id} onClick={handleReserve}>Reserve Rocket</button>
          )}
      </div>
    </div>
  );
};

RocketCard.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    active: PropTypes.bool,
    flickr_images: PropTypes.string,
    description: PropTypes.string,
    rocket_name: PropTypes.string,
  }).isRequired,
};

export default RocketCard;
