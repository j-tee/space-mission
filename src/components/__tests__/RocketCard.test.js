import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import RocketCard from '../Rocket/RocketCard';

const rocket = {
  id: 'falcon1',
  rocket_name: 'Falcon 1',
  description:
      'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
  flickr_images: 'https://imgur.com/DaCfMsj.jpg',
  active: true,
};

const MockRockets = () => (
  <Provider store={store}>
    <RocketCard rocket={rocket} />
  </Provider>
);
test('Check if renders properly', () => {
  const tree = renderer.create(<MockRockets />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('toggle reserve rocket button', () => {
  test('should log a rocket to my profile page', () => {
    render(
      (<MockRockets />),
    );
    userEvent.click(screen.getByRole('button', { name: /cancel reservation/i }));
    expect(screen.getByText(/reserved/i)).toBeInTheDocument();
  });
});
