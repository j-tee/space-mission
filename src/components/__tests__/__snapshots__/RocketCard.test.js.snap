// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if renders properly 1`] = `
<div
  className="rocket-card"
>
  <img
    alt="rocket"
    className="image"
    src="https://imgur.com/DaCfMsj.jpg"
  />
  <div
    className="rocket-desc"
  >
    <h1
      className="title-desc"
    >
      Falcon 1
    </h1>
    <p
      className="info"
    >
      <span
        className="rocket-reserve"
      >
        Reserved
      </span>
      The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
    </p>
    <button
      className="cancel-btn btn"
      id="falcon1"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation 
    </button>
  </div>
</div>
`;
